### PR TITLE
Leech Detector: add fallback check for users with no shares

### LINKED ISSUE #1565: Leech Detector is bugging people who have folders shared already.
Nicotine+ version:  
Operating System/Distribution:
3.1.1  •  GTK 3.24.30

### Describe the bug
Leech detector PMs people who are already sharing a folder (not sure how common it is, only one person has complained so far). Maybe the plugin is impatient and doesn't wait long enough if the user has a large share folder that doesn't immediately load?

### PR DESCRIPTION
**Pull request by mathiascode**
Also remove redundant user stat request, since we're watching all queued users.

Fixes #1565

### CODE REVIEW

**Comment by mathiascode** [comment]
Another thing we need to do for Nicotine+ 3.3.0 is to delay download queue requests on startup until the compressed share messages are ready. This should fix any remaining "no files shared" false positives Nicotine+ users with large shares might be experiencing.

File: pynicotine/plugins/leech_detector/__init__.py [hunk 1]
```diff
@@ -127,21 +116,18 @@ def user_stats_notification(self, user, stats):
...
-            )
-            # TODO: Implement alternative fallback method (num_files | num_folders) from a Browse Shares request
-
-        if num_files <= 0 and num_folders <= 0:
-            # SoulseekQt only sends the number of shared files/folders to the server once on startup (see Issue #1565)
-            self.log("User %s seems to have zero files and no public shared folder, the server could be wrong.", user)
+        if (num_files <= 0 or num_folders <= 0) and self.probed_users[user] != "requesting_fallback":
+            # SoulseekQt only sends the number of shared files/folders to the server once on startup.
+            # Verify user's actual number of files/folders.
+            self.log("User %s has no shared files according to the server, verifying if this is true…", user)
```
Comment (slook) [inline pynicotine/plugins/leech_detector/__init__.py:126]: "verifying" -> "requesting shares list to verify"

File: pynicotine/plugins/leech_detector/__init__.py [hunk 2]
```diff
@@ -127,21 +116,18 @@ def user_stats_notification(self, user, stats):
...
+
+            self.probed_users[user] = "requesting_fallback"
```
Comment (slook) [inline pynicotine/plugins/leech_detector/__init__.py:128]: `"requesting_fallback"` -> `"requesting_list"`

File: pynicotine/plugins/leech_detector/__init__.py [hunk 3]
```diff
@@ -85,19 +85,8 @@ def loaded_notification(self):
...
-        self.probed_users[user] = "requesting"
-
-        if user in self.core.watched_users:
-            self.core.request_user_stats(user)
-        else:
-            self.core.watch_user(user)
-
-        self.log("Getting statistics from the server for new user %s…", user)
+        if user not in self.probed_users:
+            self.probed_users[user] = "requesting"
```
Comment (slook) [inline pynicotine/plugins/leech_detector/__init__.py:89]: `"requesting"` -> `"requesting_stats"`

File: pynicotine/plugins/leech_detector/__init__.py [hunk 4]
```diff
@@ -127,21 +116,18 @@ def user_stats_notification(self, user, stats):
                          (user, num_files, num_folders))
             return
 
-        if self.probed_users[user] != "requesting":
+        if self.probed_users[user] not in {"requesting", "requesting_fallback"}:
```
Comment (slook) [inline pynicotine/plugins/leech_detector/__init__.py:119]: `if not self.probed_users[user].startswith("requesting"):`

**Review by slook — CHANGES_REQUESTED** [review]

**Pull request merged**